### ISSUE TITLE
[BUG] - Overlapping text of `unit` and `value` in `Textfield Wrapper` when in scaled mode

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#2763]
### Porsche Design System version

3.6.1

### What browsers are you seeing the problem on?

all

### Browser version

all

### Example

<img width="1153" alt="Bildschirmfoto 2023-09-07 um 13 51 29" src="https://github.com/porsche-design-system/porsche-design-system/assets/40788023/6254eeb4-9c95-4cb8-99a6-455ea298c449">

### Steps to reproduce the behavior

1. Go to https://designsystem.porsche.com/latest/components/text-field-wrapper/examples#type-number
2. Scale text up to 200%
3. See error

### Expected behavior

- [ ] `unit` and `value` text is not overlapping in case text is scaled up to 200%

### Notes
- Unit `ch` might be much better for this use case than calculated pixels, used in `packages/components/src/components/text-field-wrapper/text-field-wrapper-utils.ts`, e.g. `calc(samePaddingThanOnTheInputEnd + amountOfUnitChars**ch** + someGapBetweenUnitAndValue)`
- https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units

**Comment by michaelworm** [comment]
The main issue is done by using `ch` as value. After discussing with @denyo we decided to rebuild the case for the unit/counter to get the elements side by side and fake the unit/counter to look like an input so we get the advantages of the browser calculating the spacing between the elements by using a grid (which was already implemented, but used to place the unit/counter on top of the input which then needs a calculated spacing depending on the unit/counter text)). I am nearly finished, only hover/focus is missing and adjusting the tests again.

**Comment by michaelworm** [comment]
It is not possible to style the unit element based on hover/input states from the input via CSS since the input is a slotted element which is not part of the shadow DOM. It would require Javascript to set classes based on input's states which we don't want to do for now after discussing with @denyo and @o-hook.

**Comment by o-hook** [comment]
@stephanschroeter 
Due to the fact that the proposed CSS-only solution with `ch` unit leads to other issues (even in none-scaled mode) and that other JS-based solutions (e.g. intersection-observer) feel like "using a sledgehammer to crack a nut", I would recommend to mark this as "**won't fix**" and wait until we develop an enclosed `Textfield` component where we have more possibilities of styling.

**Comment by stephanschroeter** [comment]
Already fixed.
